### DISCUSSION TITLE
Emit JS Destructuring Syntax

### DISCUSSION THREAD

**Discussion by martinklepsch** [clavascript/clavascript#485] [Ideas]
Hi :wave:

Some tools like [playwright](https://playwright.dev/docs/test-fixtures#built-in-fixtures) (popular for UI testing) do some analysis on how functions destructure their arguments to automatically apply fixtures. While this is probably a bit of a questionable design choice it made me wonder if Squint could emit CLJS destructuring as JS destructuring forms? If that makes sense? 

Not sure if its worth it but could be interesting to consider so I thought I'd share :) 

Thanks for all your awesome work on this @borkdude 🙌 

Edit: [Vitest](https://vitest.dev/guide/test-context#fixture-initialization) also seems to use this pattern.

**Comment by borkdude** [comment]
Interesting idea, is this only about keys destructuring?

**Reply by martinklepsch** [reply]
If you mean as opposed to array destructuring then yes, I think playwright only makes use of key destructuring.

**Reply by borkdude** [reply]
I'll have a look on Monday!

**Comment by martinklepsch** [comment]
This was shipped with squint-cljs 0.7.98, here's an example using it to trigger [playwright fixtures](https://playwright.dev/docs/test-fixtures#built-in-fixtures)

```clj
(require '["@playwright/test" :refer [chromium expect test]])

(def describe (.-describe test))

(def base-uri "http://localhost:3000")

(describe "playwright meets squint"
  (fn []
    (test "has title"
      (fn ^:async title-test
        [^js {:keys [page]}]
        (js/await (.goto page base-uri))
        (js/await (.toHaveTitle (expect page) #"Dynogee"))))))
```

https://github.com/squint-cljs/squint/issues/488
